tests/engine/processes/test_builder.py

```diff
@@ -142,6 +152,25 @@ def test_builder_inputs():
     assert builder._inputs(prune=True) == {'namespace': {'nested': {'bird': []}}}
 
 
+def test_builder_populate_defaults_false():
+    """Test that namespaces with populate_defaults=False are not added to builder until accessed."""
+    # An empty builder should not have the namespace with populate_defaults=False
+    builder = PopulateDefaultsProcess.get_builder()
+    assert 'optional_ns' not in dict(builder)
+    assert builder._inputs(prune=False) == {'metadata': {}}
```

Comment:
Adding a note for clarity, as that confused me by manually breakpointing my way through the test ^^
```suggestion
    # Note that the 'metadata' dict does not contain 'options' here, as that is added in the
    # CalcJob interface, while here we are testing on a `Process` only
    assert builder._inputs(prune=False) == {'metadata': {}}
```